src/snowflake/snowpark/table_function.py

```diff
@@ -233,3 +251,43 @@ def get_column_names_from_plan(plan: SnowflakePlan) -> List[str]:
         new_cols = [Column(col)._named() for col in new_cols]
 
     return old_cols, new_cols
+
+
+def _get_cols_after_explode_join(
+    func: _ExplodeFunctionCall, plan: SnowflakePlan
+) -> List:
+    explode_col_type = None
+    for attr in plan.output:
```

Comment:
I'm curious if it worths introducing a dict in `plan` for this type search. Maybe this search can be pushed into `plan`, and `plan` can build a dict lazily such that the next search is a lookup if `explode` is called multiple times? Or is this an overkill?

Reply:
`plan` is instance of `SnowflakePlan` with has `output` as `cached_property` decorator in its definition so I think we don't need to add more optimization.

Reply:
`output` itself should be fine, my worry is about this for-loop in which we iterate over all names to find the name match which is O(n). If we have a dict then it would be O(1).